embeddings/evaluator/sequence_labeling_evaluator.py

```diff
@@ -28,42 +29,45 @@ class TaggingScheme(str, Enum):
 
     def __init__(
         self,
+        return_input_data: bool = True,
         evaluation_mode: EvaluationMode = EvaluationMode.CONLL,
         tagging_scheme: Optional[TaggingScheme] = None,
     ) -> None:
-        super().__init__()
-        self.metric = self._get_metric(evaluation_mode, tagging_scheme)
+        super().__init__(return_input_data)
+        self.evaluation_mode = evaluation_mode
+        self.tagging_scheme = tagging_scheme
 
     def _get_metric(
         self,
-        evaluation_mode: EvaluationMode,
-        tagging_scheme: Optional[TaggingScheme] = None,
     ) -> Union[HuggingFaceMetric, UnitSeqevalMetric]:
-        if evaluation_mode in SequenceLabelingEvaluator.SEQEVAL_EVALUATION_MODES:
-            if evaluation_mode == "strict" and not tagging_scheme:
+        if self.evaluation_mode in SequenceLabelingEvaluator.SEQEVAL_EVALUATION_MODES:
+            if self.evaluation_mode == "strict" and not self.tagging_scheme:
                 raise ValueError("Tagging scheme must be set, when using strict evaluation mode!")
-            elif evaluation_mode == "conll" and tagging_scheme:
+            elif self.evaluation_mode == "conll" and self.tagging_scheme:
                 raise ValueError("Tagging scheme can be set only in strict mode!")
             return HuggingFaceMetric(
                 name="seqeval",
                 compute_kwargs={
-                    "mode": evaluation_mode if evaluation_mode == "strict" else None,
-                    "scheme": tagging_scheme,
+                    "mode": self.evaluation_mode if self.evaluation_mode == "strict" else None,
+                    "scheme": self.tagging_scheme,
                 },
             )
-        elif evaluation_mode == "unit":
+        elif self.evaluation_mode == "unit":
             return UnitSeqevalMetric()
         else:
             raise ValueError(
-                f"Evaluation mode {evaluation_mode} not supported. Must be one of "
+                f"Evaluation mode {self.evaluation_mode} not supported. Must be one of "
                 f"[unit, conll, strict]."
             )
 
-    @property
     def metrics(
         self,
-    ) -> Sequence[Metric[Union[List[Any], nptyping.NDArray[Any], torch.Tensor], Dict[Any, Any]]]:
-        return [self.metric]
+    ) -> Dict[str, Metric[Union[List[Any], nptyping.NDArray[Any], torch.Tensor], Dict[Any, Any]]]:
+        return {self.get_metric_name(self.evaluation_mode, self.tagging_scheme): self._get_metric()}
+
+    @property
+    def evaluation_results_cls(self) -> Type[SequenceLabelingEvaluationResults]:
+        return SequenceLabelingEvaluationResults
```

Comment:
do we need this property? is it use outside of the `evaluate` method inside this class? we could just simply provide the class name in the `evaluate` method `return SequenceLabelingEvaluationResults(...)`